onnxruntime/python/tools/quantization/save_activations.py

```diff
@@ -0,0 +1,156 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft, Intel Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""Utilities to run a given ONNX model, while saving input/output tensors of
+eligible operator nodes.
+
+A use case is to debug quantization induced accuracy drop. An AI engineer can
+run the original float32 model and the quantized model with the same inputs,
+then compare the corresponding activations between the two models to find
+where the divergence is. 
+
+Example Usage:
+
+.. code-block:: python
+    class ExampleDataReader(CalibrationDataReader):
+        def __init__(self):
+            ...
+        def get_next(self):
+            ...
+
+    input_data_reader = ExampleDataReader()
+
+    aug_model = aug_model_for_act_saving(path_to_onnx_model)
+    augmented_model_path = str(Path(self._tmp_model_dir.name).joinpath("augmented_model.onnx"))
+    onnx.save(
+        aug_model,
+        augmented_model_path,
+        save_as_external_data=False,
+    )
+
+    tensor_dict = run_collect_activations(augmented_model_path, data_reader)
+    # tensor_dict points to a dictionary where the keys are tensor names and each value
+    # is a list of tensors, one from each model run
```

Comment:
Prefer markdown
```suggestion
```python
    class ExampleDataReader(CalibrationDataReader):
        def __init__(self):
            ...
        def get_next(self):
            ...

    input_data_reader = ExampleDataReader()

    aug_model = aug_model_for_act_saving(path_to_onnx_model)
    augmented_model_path = str(Path(self._tmp_model_dir.name).joinpath("augmented_model.onnx"))
    onnx.save(
        aug_model,
        augmented_model_path,
        save_as_external_data=False,
    )

    tensor_dict = run_collect_activations(augmented_model_path, data_reader)
    # tensor_dict points to a dictionary where the keys are tensor names and each value
    # is a list of tensors, one from each model run
```
```

Reply:
```
str(Path(self._tmp_model_dir.name).joinpath("augmented_model.onnx"))
```
->
```
str(Path(self._tmp_model_dir.name) / "augmented_model.onnx")
```

nit: consider

Reply:
Path.joinpath() uses "\" in Windows and "/" in Linux/Unix/Mac right?

Reply:
I think Path is os aware:

```
>>> Path("abc") / "def"
WindowsPath('abc/def')
>>> str(Path("abc") / "def")
'abc\\def'
```

Reply:
`/` is simply a nice way of calling joinpath

Inside the pathlib implementation:

```
    def joinpath(self, *args):
        """Combine this path with one or several arguments, and return a
        new path representing either a subpath (if all arguments are relative
        paths) or a totally different path (if one of the arguments is
        anchored).
        """
        return self._make_child(args)

    def __truediv__(self, key):
        try:
            return self._make_child((key,))
        except TypeError:
            return NotImplemented
```

Reply:
it has operator overloading, wow

Reply:
fixed